purchase_order_line_deep_sort/models/purchase_order.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Tecnativa - Vicent Cubells <vicent.cubells@tecnativa.com>
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from odoo import api, fields, models
+from .res_company import SORTING_CRITERIA, SORTING_DIRECTION
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    line_order = fields.Selection(
+        selection=SORTING_CRITERIA,
+        string='Sort Lines By',
+        default=lambda self: self.env.user.company_id.default_po_line_order,
+    )
+    line_direction = fields.Selection(
+        selection=SORTING_DIRECTION,
+        string='Sort Direction',
+        default=lambda self:
+            self.env.user.company_id.default_po_line_direction,
+    )
+
+    @api.onchange('line_order')
+    def onchange_line_order(self):
+        if not self.line_order:
+            self.line_direction = False
+
+    @api.multi
+    def _sort_purchase_line(self):
+        def resolve_subfields(obj, line_order):
+            subfields = line_order.split('.')
+            res = obj
+            for subfield in subfields:
+                res = getattr(res, subfield)
+            return res
+
+        if not self.line_order and not self.line_direction:
+            return
+        reverse = self.line_direction == 'desc'
+        sequence = 0
+        sorted_lines = self.order_line.sorted(
+            key=lambda p: resolve_subfields(p, self.line_order),
+            reverse=reverse,
+        )
+        for line in sorted_lines:
+            sequence += 10
+            line.sequence = sequence
```

Comment:
@cubells @pedrobaeza You cannot update a recordset like that in a write() or a create(). You are not sure of the behaviour !

Reply:
Why do you always review after being months here and it's merged? Any way, this is something correct, and I don't see how can you improve this. The only optimization is to check first if the value is the same, but that's not usual to happen several times.

Reply:
Don't have antennas like you. Depends on activity. But I'd prefer letting a comment to make author warned than nothing even if merged.

Reply:
Hehe, OK, no particular problem, but for avoiding something to be available in repo if you don't consider it correct, but I still don't see the problem you are pointing.

Reply:
It's not recommended to update a recordset like `recordset.property = value' in a write() or create() as you are not sure what could be re-triggered. You also are not sure that changes will be commited as all the changes are done in the super().

Reply:
That's a good general advice, but this time, everything is very constrained to the specific change of some values, and as this has to be done this way for covering all cases from write/create in the purchase.order and purchase.order.line. Little optimizations can be done for passing the `sequence` value in the same dictionary, but they are not possible in other cases and it will complicate unnecessary the code.

Reply:
So the summary is:

1. There's no possible problem with infinite loops.
2. Maybe in some cases you can optimize an SQL call.

My opinion is that 2 doesn't worth right now, but if you think it is, please propose a way to avoid that calls.